live-objects/src/main/kotlin/io/ably/lib/objects/Helpers.kt

```diff
@@ -0,0 +1,43 @@
+package io.ably.lib.objects
+
+import io.ably.lib.realtime.CompletionListener
+import io.ably.lib.types.ErrorInfo
+import io.ably.lib.types.ProtocolMessage
+import kotlinx.coroutines.CompletableDeferred
+
+internal suspend fun LiveObjectsAdapter.sendAsync(message: ProtocolMessage) {
```

Comment:
it's better to use `suspendCancellableCoroutine` to convert callback to suspended function (it's recommended way)

Reply:
updated